### ISSUE TITLE
Mystcraft craftable items not showing in JEI, not able to craft

### ISSUE THREAD

**Issue by trsblur** [Darkosto/SkyFactory4Issues#910]
[latest.log](https://github.com/Darkosto/SkyFactory4Issues/files/3167807/latest.log)
## Bug Report

CampactSky - Mystcraft items not showing in JEI even after unlocking it through prestige, or creating a new non-prestige compactsky world. Items are also uncraftable. Mystcraft works fine on all other map types I have tried (tree+void, default, tutorial, skylands, underwater) 

## Expected Behaviour

Craft Mystcraft items and look up recipes upon unlocking the mod.

## Steps to Reproduce (for bugs)
1.create a compactsky world with prestige enabled
2.unlock mystcraft with prestige
3.search @mystcraft in JEI

## Logs
* Client/Server Log:
* Crash Log:

## World Information

* Preset:compactsky

* Prestige: yes

* Modpack Version world created in:4.0.1 and 4.0.5

* Additional Content Installed:

## Client Information

* Modpack Version:4.0.5
* Java Version:latest
* Launcher Used:twitch
* Memory Allocated:8GB
* Server/LAN/Single Player:Single Player
* Optifine Installed:no
* Shaders Enabled:no

**Comment by sk2048** [comment]
Mystcraft is purposefully disabled in CompactMachines and Bedrock world types as it defeats the challenge of these maps. We should probably document this better.

**Comment by MuteTiefling** [comment]
If this is intentional, it's also impossible to complete the nuclearcraft fusion challenge. This requires a minimum of a size 21 toroid D-T fusion reactor to generate 3.36 million rf/t. That's a 49x49 block structure.

Incidentally, not knowing mystcraft was meant to be disabled, I was able to craft everything and create an age in a compactsky world... I've since completed the fusion reactor as well.
